pyproject.toml

```diff
@@ -17,3 +17,41 @@ exclude_lines = [
     "if __name__ == .__main__.:",
     "if TYPE_CHECKING:",
 ]
+
+[tool.ruff]
+exclude = [
+    "__init__.py",
+]
+ignore = [
+    "D100",  # Missing docstring in public module
+    "D205",  # 1 blank line required between summary line and description
+    "N802",
+    "N803",
+    "N806",
+    "N812",
+    "N813",
+    "N815",
+    "N816",
+]
+line-length = 110
+select = [
+    "D",  # pydocstyle
+    "E",  # pycodestyle
+    "F",  # pyflakes
+    "N",  # pep8-naming
+    "W",  # pycodestyle
+    "UP",  # pyupgrade
+    "C4",  # comprehensions
+    "B",  # bugbear
+    "SIM",  # simplify
+]
+target-version = "py311"
+extend-select = [
+    "RUF100", # Warn about unused noqa
```

Comment:
> Looks like linting with ruff when running scons in the test.yaml fails because of incompatible versions. 

This is the one that causes problems with version mismatches. Easiest fix is to remove the RUF100 and then add noqas that are compatible with both.